### PR TITLE
Add .verify() and .resetCount()

### PR DESCRIPTION
**Pull request by mroderick**
Ref discussion in #47

* Improve `referee.count` so that it can't be written to directly.
* Add `referee.verify()` which will verify that count is not zero, and optionally is expected value
* Add `referee.resetCount()`, which predictably resets the count to zero

#### How to verify
1. Check out this branch
2. `npm install`
3. `npm test`

#### Checklist for author

- [x] `npm run lint` passes
- [ ] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523).
- [ ] Write documentation
- [ ] Add tests

### CODE REVIEW

File: lib/create-verify.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    function verify(expected) {
+        if (
+            typeof expected !== "undefined" &&
+            (typeof expected !== "number" || expected < 1)
+        ) {
+            throw new TypeError("expected argument must be a number >= 1");
+        }
+
+        var exp = expected || 1;
+        if (referee.count === exp) {
```
[the remark this answers is not in the document]
Comment (mroderick) [reply lib/create-verify.js]: Indeed. I've pushed a commit to fix this

[the remark this answers is not in the document]
Comment (mantoni) [reply lib/create-verify.js]: Nice, thank you :)

**Comment by mantoni** [comment]
Very nice work. I’m wondering if `verify()` should be resetting the count?

**Comment by mroderick** [comment]
> Very nice work. I’m wondering if `verify()` should be resetting the count?

I was wondering about that myself, but wasn't sure. I've pushed yet another commit™️ that makes it do that.

File: lib/create-verify.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            );
+        }
+
+        if (referee.count === 0) {
+            throw new Error(
+                "Expected assertion count to be at least 1, but was 0"
+            );
+        }
+
+        referee.resetCount();
```
Comment (mantoni) [inline lib/create-verify.js]: There is another edge case: If a test fails due `referee.count` not matching `expected`, then `count` is not reset to zero, potentially leaving an unwanted / unexpected side effect to a following test.

Comment (mroderick) [reply lib/create-verify.js]: Hmm ... I considered that it shouldn't "destroy the evidence", but of course we will be running tests after this.

Stateful libraries are messy.

I'll fix this

**Comment by mroderick** [comment]
Apologies for force pushing, I thought we were finished with this branch.

I'll try not to put in unrelated commits (travis.yml) in the future, so we can just use "Squash and merge"

**Comment by mroderick** [comment]
Pushed another commit to always reset count on verify

**Comment by mantoni** [comment]
No worries. This looks good to me now 👍

**Comment by mantoni** [comment]
I gave this a shot in a project that uses `sinon-referee` by linking `referee` locally. Apart from [the one compatibility issue mentioned here](https://github.com/sinonjs/referee/commit/f30acf6861701975bb3ecc8895557a258d8b42fa#commitcomment-31255059) the latest changes don't seem to cause any issues in this context.

I'm using these new APIs with Mocha like this now:

```js
const { assert, sinon, resetCount, verify } = require('@sinonjs/referee-sinon');

describe('resetCount and verify', () => {

  before(() => {
    resetCount();
  });

  afterEach(() => {
    sinon.restore();
    verify();
  });

  it('causes a failure due to missing assertions', () => {
    // Nothing here
  });

});
```

Note that `resetCount` must be called in `before` because other test files that do not call `verify()` (yet) increment the assertion count and then `verify()` doesn't catch it here. Like you said ... stateful libraries – but I like it anyway 😄
